app/models/source.rb

```diff
@@ -9,14 +9,13 @@ class UnknownRankable < StandardError; end
   validates :phone_number_rank, presence: true, uniqueness: true
   validates :website_rank, presence: true, uniqueness: true
 
-  def rank_for(rankable)
-    case rankable
-    when Email then email_rank
-    when Location then location_rank
-    when OrganizationName then organization_name_rank
-    when PhoneNumber then phone_number_rank
-    when Website then website_rank
-    else raise UnknownRankable
-    end
+  def rank_for(type)
+    {
+      email_rank: email_rank,
+      location_rank: location_rank,
+      organization_name_rank: organization_name_rank,
+      phone_number_rank: phone_number_rank,
+      website_rank: website_rank
+    }[type] || raise(UnknownRankType)
```

Comment:
I like `}[type]`, but it feels a little cryptic. Perhaps consider pulling out the hash into a local.